libevmasm/Assembly.h

```diff
@@ -157,6 +157,8 @@ class Assembly
 	std::vector<size_t> decodeSubPath(size_t _subObjectId) const;
 	size_t encodeSubPath(std::vector<size_t> const& _subPath);
 
+	bool isCreation() const { return m_creation; }
```

Comment:
What about changing this to "executedOnce"?

Reply:
The effect in the optimizer is not only adjusting the runs parameter, but also stuff like
``m_params.isCreation ? GasCosts::txDataNonZeroGas(m_params.evmVersion) : GasCosts::createDataGas``
in the constant optimizer calculations...

Reply:
Yeah but as I said, for tx data gas, it should be relevant whether it is a top-level object.

Reply:
Yeah, but right now the flag does both, so it's really a "creation mode" flag so far...

If we want to additionally consider whether something is a top-level object, then we run into the same problem again: the compiler would output different bytecode for ``C`` compiled in isolation compared to ``C`` used from another contract ``D``... what we really need is per-contract optimizer settings for this, don't we?